account_invoice_auto_send_by_email/models/account_move.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def cron_send_email_invoice(self):
+        invoices = self.search(self._email_invoice_to_send_domain())
+        for invoice in invoices:
+            description = "Send invoice {} by email".format(invoice.name)
+            invoice.with_delay(description=description)._execute_invoice_sent_wizard()
```

Comment:
Maybe it would be interesting to check if the invoice is already being processed with a boolean field "sending_in_progress"? (like its done in account_invoice_mass_sending)

Reply:
Good point.
I have just added a check at the beginning of the function executed by the job (cf. fixup commit)

Reply:
And to avoid multiple jobs to be spawn for the same invoice, you can use `identify_key`:
```python
from odoo.addons.queue_job.job import identity_exact
[...]
invoice.with_delay(identity_key=identity_exact)._execute_invoice_sent_wizard()
```

Reply:
Good to know thanks, I'll look into it.

Reply:
Adding an identity key will prevent creating a new job if some other job with the same key have NOT RUN yet. Which in this case is not so important because of the daily schedule.
The change I made was at the beginning of the job to check that the invoice has not yet been sent, to prevent an invoice being sent multiple times to the customer.